### DISCUSSION TITLE
Calling opengl functions causes crash

### DISCUSSION THREAD

**Discussion by alexgu754** [google/filament#5082] [Q&A] [answered]
as the title says, calling functions like glCreateShader(...); causes a crash

in the console it prints "FEngine resolved backend: OpenGL" during startup, so opengl must work somehow

https://imgur.com/SBOuvYB
pic related is my failed attempt

init is called in FilamentApp::get().run setup lambda
the intent is to compute mipmaps on the gpu at runtime, because mipgen takes a couple minutes just to process 1 texture

if anyone could point me in the right direction, that would be great

**Comment by romainguy** [comment, marked as answer]
Filament does not expose internal graphics APIs to apps. The context is locked to an internal thread that you don't have access to. Filament offers an API on Texture to generate the mip maps though.

**Reply by alexgu754** [reply]
what is this other api?
could you tell me please.

mipgen is painfully slow. each mip is generated from the source texture. 
no multi threading, no avx/sse intrinsics. 
if its a 4k or 8k texture, it takes aeons

edit: found it Texture::generateMipmaps()

**Reply by romainguy** [reply]
https://github.com/google/filament/blob/main/filament/include/filament/Texture.h#L474
